auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -135,22 +218,46 @@
      * @param root the root path of the identity store
      * @param nameRewriter the name rewriter to apply to looked up names
      * @param levels the number of levels of directory hashing to apply
-     * @param encoded whether identity names should be BASE32 encoded before using as filename
+     * @param encoded whether identity names should be BASE32 encoded before using as filename (only applies if the security realm is unencrypted)
      * @param hashCharset the character set to use when converting password strings to a byte array. Uses UTF-8 by default.
      * @param hashEncoding the string format for the hashed passwords. Uses Base64 by default.
+     * @param providers The providers supplier
+     * @param secretKey the SecretKey used to encrypt and decrypt the security realm (if {@code null}, the security realm will be unencrypted)
+     * @param privateKey the PrivateKey used for writing the signature of the identities in security realm (if {@code null}, the security realm will not have integrity enabled)
+     * @param publicKey the PublicKey used for verifying the signature of the identities in security realm (if {@code null}, the security realm will not have integrity enabled)
      */
-    public FileSystemSecurityRealm(final Path root, final NameRewriter nameRewriter, final int levels, final boolean encoded, final Encoding hashEncoding, final Charset hashCharset, Supplier<Provider[]> providers) {
+    public FileSystemSecurityRealm(final Path root, final NameRewriter nameRewriter, final int levels, final boolean encoded, final Encoding hashEncoding, final Charset hashCharset, Supplier<Provider[]> providers, final SecretKey secretKey, final PrivateKey privateKey, final PublicKey publicKey) {
```

Comment:
Just thinking about this one some more. Since the intent going forwards is for users to make use of the builder to create a FileSystemSecurityRealm, perhaps "public" could be removed from this constructor?

Reply:
Makes sense to me. Removed `public`